rd-net/RdFramework/Impl/Serializers.cs

```diff
@@ -277,14 +277,13 @@ public T Read<T>(SerializationCtx ctx, UnsafeReader reader, [CanBeNull] CtxReadD
         return default(T);
       var size = reader.ReadInt();
 
-      CtxReadDelegate<object> ctxReadDelegate;
-      if (!myReaders.TryGetValue(typeId, out ctxReadDelegate))
+      if (!TryGetReader(typeId, out var ctxReadDelegate))
       {
         if (unknownInstanceReader == null)
         {
           myRegistrar?.TryRegister(typeId, this);
           myRegistrar?.TryRegister(typeof(T), this);
-          if (!myReaders.TryGetValue(typeId, out ctxReadDelegate))
+          if (!TryGetReader(typeId, out ctxReadDelegate))
           {
             var realType = myTypeMapping.SingleOrDefault(c => Equals(c.Value, typeId)); //ok because it's rarely needed
```

Comment:
myTypeMapping is read without a lock.
Additionally, myWriter is read without a lock on line 341